viper.go

```diff
@@ -260,47 +261,70 @@ func (v *Viper) OnConfigChange(run func(in fsnotify.Event)) {
 
 func WatchConfig() { v.WatchConfig() }
 func (v *Viper) WatchConfig() {
+	initWG := sync.WaitGroup{}
+	initWG.Add(1)
```

Comment:
If it's just a wait group of one, couldn't this just be synchronous?

Reply:
actually, no, it's slightly more complex: there's a first go routine (from https://github.com/xcoulon/viper/blob/242f4890f5f6239217fa5c45bb6de3b4b5509d3b/viper.go#L266) that takes care about the initialization of the `watch` go routine that will look after the FS changes (this latter routine has an infinite loop, so it continuously monitors the changes). But that's the `initialization` go routine that is in charge of starting the FSNotifier and closing it once the watcher routine is done, so this routine needs to keep running. The main routine only blocks/waits until the initialization is complete, but the `initialization` rountine keeps running nonetheless. I hope this makes sense...

In the past, the code was using channels to block (with something like `<- done`), which I replaced with `sync.WaitGroup`, but maybe that's too complicated/overrated ?